rest/api_collections_test.go

```diff
@@ -269,6 +269,125 @@ func TestMultiCollectionDCP(t *testing.T) {
 	// require.NoError(t, rt.WaitForDoc(docID))
 }
 
+func TestMultiCollectionChannelAccess(t *testing.T) {
+	base.TestRequiresCollections(t)
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyAll)
+
+	if !base.TestUseXattrs() {
+		t.Skip("Test relies on import - needs xattrs")
+	}
+
+	tb := base.GetTestBucket(t)
+	defer tb.Close()
+
+	scopesConfig := GetCollectionsConfig(t, tb, 3)
+	dataStoreNames := GetDataStoreNamesFromScopesConfig(scopesConfig)
+	c1SyncFunction := `function(doc) {access(doc.username, [doc.grant1,doc.grant2]);
+						channel(doc.chan);
+				}`
+
+	scope := dataStoreNames[0].ScopeName()
+	collection1 := dataStoreNames[0].CollectionName()
+	collection2 := dataStoreNames[1].CollectionName()
+	collection3 := dataStoreNames[2].CollectionName()
+
+	scopesConfig[scope].Collections[collection1] = CollectionConfig{SyncFn: &c1SyncFunction}
+	scopesConfig[scope].Collections[collection2] = CollectionConfig{SyncFn: &c1SyncFunction}
+
+	rtConfig := &RestTesterConfig{
+		CustomTestBucket: tb,
+		PersistentConfig: true,
+	}
+
+	rt := NewRestTesterMultipleCollections(t, rtConfig, 3)
+	defer rt.Close()
+	scopesConfigString, err := json.Marshal(scopesConfig)
+	resp := rt.SendAdminRequest("PUT", "/db/", fmt.Sprintf(
+		`{"bucket": "%s", "num_index_replicas": 0, "enable_shared_bucket_access": %t, "use_views": %t, "scopes":%s}`,
+		tb.GetName(), base.TestUseXattrs(), base.TestsDisableGSI(), scopesConfigString))
+	RequireStatus(t, resp, http.StatusCreated)
+
+	userPayload := `{
+		"password":"letmein",
+		"collection_access": {
+			"%s": {
+				"%s": {
+					"admin_channels":%s
+				}
+			}
+		}
+	}`
+
+	// Create a few users with access to various channels via admin grants
+	resp = rt.SendAdminRequest("PUT", "/db/_user/userA", fmt.Sprintf(userPayload, scope, collection1, `["A"]`))
+	RequireStatus(t, resp, http.StatusCreated)
+	resp = rt.SendAdminRequest("PUT", "/db/_user/userB", fmt.Sprintf(userPayload, scope, collection1, `["B"]`))
+	RequireStatus(t, resp, http.StatusCreated)
+	resp = rt.SendAdminRequest("PUT", "/db/_user/userAB", fmt.Sprintf(userPayload, scope, collection1, `["A","B"]`))
+	RequireStatus(t, resp, http.StatusCreated)
+
+	// Write docs to both collections in various channels
+	resp = rt.SendAdminRequest("PUT", "/{{.keyspace1}}/testDocBarA", `{"username": "userA", "grant1": "A", "chan":["A"]}`)
+	RequireStatus(t, resp, http.StatusCreated)
```

Comment:
RE: previous comment about removing dynamic grants from this test - we're granting `userA` access to channel `A` in two different ways between these lines. We would be unable to tell which one is correct/working and we're trying to test the admin_channels grants.

Reply:
addressed in [14b870c](https://github.com/couchbase/sync_gateway/pull/6012/commits/14b870cf30370886162344a52d2606ce581b002a)